crates/rust-analyzer/src/bin/main.rs

```diff
@@ -3,6 +3,10 @@
 //! Based on cli flags, either spawns an LSP server, or runs a batch analysis
 
 #![warn(rust_2018_idioms, unused_lifetimes, semicolon_in_expressions_from_macros)]
+#![cfg_attr(feature = "in-rust-tree", feature(rustc_private))]
+#[cfg(feature = "in-rust-tree")]
```

Comment:
This either expands to:

```rust
#[allow(unused_extern_crates)]
extern crate rustc_driver;
```

or to nothing, right? At least it still works without the feature enabled.

I _should_ know, but..

Reply:
yes, the `cfg` configures the whole annotated item, including its attributes